lib/rules/at-rule-allowed-list/index.js

```diff
@@ -17,23 +17,19 @@ const meta = {
 	url: 'https://stylelint.io/user-guide/rules/list/at-rule-allowed-list',
 };
 
-/** @type {import('stylelint').Rule} */
+/** @type {import('stylelint').Rule<string | string[]>} */
 const rule = (primary) => {
-	// To allow for just a string as a parameter (not only arrays of strings)
-	const primaryValues = [primary].flat();
```

Comment:
[note] `validateOptions` allows both a single value and an array, so it's better to call `.flat()` after the validation.